src/pal/src/safecrt/internal_securecrt.h

```diff
@@ -23,6 +23,17 @@
 #ifndef _INC_INTERNAL_SECURECRT
 #define _INC_INTERNAL_SECURECRT
 
+#if defined(__linux__) && !defined(__GLIBC__)  // e.g. OS with musl-libc
```

Comment:
Whcih version number of Ubuntu is that? I have tested these patches on several operating erating systems + dotnet CI passing, and found this as a most appropriate balance. If this macro was defined on any Linux flavor other than those rely on non-GNU-libc, we would get `type redefinition` fatal error from compiler at 3-5% of the build on all the OSes supported by CoreCLR. To test, remove this condition and compile on Linux/BSD/OSX. There is a reason we have `#if defined(__linux__)` before `!defined(__GLIBC__)` and this is the known way of identifying Alpine.